### PR TITLE
Fix: update the patch trait doc

### PR DESCRIPTION
**Pull request by FogDong**
Signed-off-by: FogDong <dongtianxin.tx@alibaba-inc.com>

/cc @wonderflow

### CODE REVIEW

File: i18n/zh/docusaurus-plugin-content-docs/current/platform-engineers/traits/patch-trait.md [hunk 1]
```diff
@@ -241,6 +262,9 @@ spec:
       - name: retainkeys-demo-ctr
         image: nginx
 ```
+
+#### 2. 使用 `+patchStrategy=retainKeys` 注解
```
Comment (wonderflow) [inline i18n/zh/docusaurus-plugin-content-docs/current/platform-engineers/traits/patch-trait.md]: are we missing something here?

File: i18n/zh/docusaurus-plugin-content-docs/current/platform-engineers/traits/patch-trait.md [hunk 2]
```diff
@@ -354,7 +378,7 @@ spec:
         		// +patchKey=name
         		containers: [{
         			name: context.name
-        			// +patchKey=name
+        			// +patchStrategy=retainKeys
```
Comment (wonderflow) [inline i18n/zh/docusaurus-plugin-content-docs/current/platform-engineers/traits/patch-trait.md:440]: typo?

File: docs/platform-engineers/traits/patch-trait.md [hunk 3]
```diff
@@ -48,15 +48,29 @@ spec:
...
         }
 
+        patchOutputs: {
+          service: {
+            metadata: annotations: {
+              "patched-by": traits
+            }
+          }
+        }
+
```
Comment (wonderflow) [inline docs/platform-engineers/traits/patch-trait.md]: let's make a separate example and make it clear about the results

File: i18n/zh/docusaurus-plugin-content-docs/current/platform-engineers/traits/patch-trait.md (hunk 2 above)
Comment (FogDong) [reply i18n/zh/docusaurus-plugin-content-docs/current/platform-engineers/traits/patch-trait.md:440]: We should use `retainKeys` here, the original example has some problems.

File: docs/platform-engineers/traits/patch-trait.md [hunk 4]
```diff
@@ -86,7 +97,64 @@ spec:
...
+  name: patch-annotation
+spec:
+  appliesToWorkloads:
+    - deployments.apps
+  podDisruptive: true
+  schematic:
+    cue:
+      template: |
+        patchOutputs: {
+          service: {
```
Comment (wonderflow) [inline docs/platform-engineers/traits/patch-trait.md]: ingress

File: docs/platform-engineers/traits/patch-trait.md [hunk 5]
```diff
@@ -86,7 +97,64 @@ spec:
...
+        image: oamdev/testapp:v1
+      traits:
+        - type: "gateway"
+          properties:
+            domain: testsvc.example.com
+            http:
+              "/": 8000
+        - type: "patch-annotation"
+          properties:
+            name: "patch-annotation-trait"
```
Comment (wonderflow) [inline docs/platform-engineers/traits/patch-trait.md:150]: patch istio labels, let's use real world example to make it more clear

File: docs/platform-engineers/traits/patch-trait.md [hunk 6]
```diff
@@ -86,7 +97,64 @@ spec:
...
+```
```
Comment (wonderflow) [inline docs/platform-engineers/traits/patch-trait.md:151]: give the result YAML  about the patched ingress yaml here.

**Review by wonderflow — APPROVED** [review]

**Pull request merged**